src/V3ExecGraph.cpp

```diff
@@ -380,12 +385,14 @@ class PackThreads final {
                 // If no tasks were added to the normal thread schedule, remove it.
                 if (schedule.mtaskState.empty()) result.erase(result.begin());
```

Comment:
This is in existing code, but `schedule` is the last element of `result`, but here you always seem to delete the first.

Reply:
This only handles the situation when we first get a multi-threaded task without normal one at the start, thus only a single empty thread schedule will be present at this time which should be removed. To simplify this, I replaced it with `clear()` operation, and adding assert in the other scheduling mode.